### ISSUE TITLE
Cannot user insert feature with database specific right

### ISSUE THREAD

**Issue by Kaldor37** [phpmyadmin/phpmyadmin#13001]
### Steps to reproduce
1. Grant only SELECT global right to the user
2. Grant INSERT right on the desired database
3. Connect with the user to the database
4. Choose a table
5. Click "Insert" button
6. Notice that all fields are readonly and functions are unavailable

### Expected behaviour
With the database specific INSERT right granted, the user should be able to INSERT actual data to the tables.

### Actual behaviour
The "Insert" button is accessible and sends to the actual page but all fields are "readonly" and the "functions" column is empty.
![rights](https://cloud.githubusercontent.com/assets/11570956/22936196/097629ce-f2d5-11e6-95f8-40b989f24573.png)
![insert](https://cloud.githubusercontent.com/assets/11570956/22936043/96fb28ea-f2d4-11e6-8984-4a2e00cf7a6d.png)

### Server configuration
**Operating system**: Debian 8

**Web server:** Apache/2.4.10 (Debian) 

**Database:** MariaDB (0.0.29-MariaDB-0+deb8u1 - (Debian))

**PHP version:** 5.6.29-0+deb8u1 

**phpMyAdmin version:** 4.6.5.2

### Client configuration
**Browser:** Firefox 51.0.1

**Operating system:** Debian 8

**Comment by jublo** [comment]
So you have these permissions set, correct?

![image](https://cloud.githubusercontent.com/assets/6348321/22954284/d90a823a-f315-11e6-86ef-6fde23b51960.png)

I can reproduce the issue you describe, on `QA_4_7` and `master`.

**Comment by jublo** [comment]
This is because the SQL command `SHOW FULL COLUMNS FROM …` returns the `select` privilege only:

![image](https://cloud.githubusercontent.com/assets/6348321/22954702/538b69b4-f318-11e6-96d2-fcca8541d895.png)

**Comment by Kaldor37** [comment]
I can confirm that this is exactly what happens.
SHOW FULL COLUMNS on any tables of the database doesn't returns the insert privilege.

**Comment by nijel** [comment]
Looks like MySQL bug: https://bugs.mysql.com/bug.php?id=76390

It seems that SHOW FULL COLUMNS is not really a good source for figuring out privileges...

**Comment by bobweil** [comment]
This bug appear also to affect updating, not just inserting.

Upon upgrading from an older version of PMA, all my users who had database-specific insert,update rights have lost their ability to type values into the edit form.  If I assign a column-specific privilege to one of these columns, then they can type in only that column.

With only the database-specific privilege, they can still edit a column in the grid view by double-clicking, but cannot type in the edit form for a single row.

**Comment by duncan3dc** [comment]
@nijel What's the status of this? Are you interested in a pr to use a different method of figuring out privileges? Or would you accept a pr to revert the feature that introduced this issue (ff1592483a9befbe93db30b766f4b38da4c775f1)?

**Comment by nijel** [comment]
This really should be fixed in MySQL. On the other side we might workaround that by using different source for per column privileges...

**Comment by ChrisHSandN** [comment]
For anyone finding this, as a workaround: if you remove all global privileges from the user (and re-add them to all databases as database-specific privileges). Its not ideal as if you create any new databases you will have to remember to add the correct privileges for each user like this.

Any chance of a fix for this? - database specific privileges not working is a fairly major bug and this has been open for 8 months now. Even just removing the readonly from the insert/update fields and letting it go through to an SQL error would be preferable.

**Comment by bobweil** [comment]
My workaround does let it go through as an SQL error.  

In phpmyadmin/libraries/insert_edit.libraries.php, near the bottom, there's a function called "PMA_userHasColumnPrivileges."  Just short circuit the body of this function by inserting a "return true;" statement.  This will let MySQL handle any authorization issues.

```
function PMA_userHasColumnPrivileges($table_column, $insert_mode)
return true;
...

```

**Comment by williamdes** [comment]
Hi everyone, is this issue still up to date ?

**Comment by jeromemnx** [comment]
> Hi everyone, is this issue still up to date ?

Hi, I have this issue on a fresh install (debian 9.9 / MariaDB 15.1 / phpmyadmin 4.6.6deb4).

**Comment by jeromemnx** [comment]
What can I do to help solving this issue ?

**Comment by williamdes** [comment]
@jeromemnx Can you apply #14993 on your local phpMyAdmin to see if it solved the issue ?

**Comment by jeromemnx** [comment]
> @jeromemnx Can you apply #14993 on your local phpMyAdmin to see if it solved the issue ?
Yes it solve the problem but it remove the readonly property also for databases where my user doesn't have insert right.
